### PR TITLE
ntpd: Make SNTP support optional

### PR DESCRIPTION
**Pull request by KloudJack**
Add configuration option for SNTP support with it being disabled by default.
Add libevent2-pthreads dependency required by SNTP only when it is enabled.

Fixes openwrt#10307
Closes openwrt#10308

Maintainer: @tripolar 
Compile tested: OpenWRT snapshot r11266-34939711a5
Run tested: ZyXEL NBG6817 w/ OpenWRT snapshot r11266-34939711a5

### CODE REVIEW

**Comment by KloudJack** [comment]
Apologies for my other closed erroneous pull request. Here I thought Github was smart enough to detect identical commits with different commit ids after a rebase, guess that's not the case...

**Comment by KloudJack** [comment]
Do you mean ntp-utils?

I agree, that's probably a better approach here; will update with a new commit.

**Comment by KloudJack** [comment]
As `--with-sntp` is implicit during configure, I chose to use a one-line conditional in the CONFIGURE_ARGS to select `--without-sntp` when ntp-utils isn't selected. This is to prevent the unnecessary libevent2 dependency for the other packages.

**Comment by feckert** [comment]
@tripolar @KloudJack could this be merged?

**Comment by KloudJack** [comment]
@tripolar the merged #10308 just fixes the sntp build dependency without actually including the sntp binary in the staging/target dirs.

This PR will include the binary while also making it optional when building ntpd, so it'd be nice if we could get this merged as well.

File: net/ntpd/Makefile [hunk 1]
```diff
@@ -146,6 +148,7 @@ define Package/ntp-utils/install
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/ntpdc/ntpdc $(1)/usr/sbin/
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/ntpq/ntpq $(1)/usr/sbin/
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/util/ntptime $(1)/usr/sbin/
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/sntp/sntp $(1)/usr/sbin/
```
Comment (feckert) [inline net/ntpd/Makefile]: You need also to check if we have compile sntp. If not then this will fail.

Comment (KloudJack) [reply net/ntpd/Makefile]: Only when ntp-utils are selected, will sntp will be built and installed. See line 112.

@tripolar requested this in this PR above.

**Pull request closed without merging**